File: LeetCode-Python/53-最大子序和.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        sum = 0
        max_sum = nums[0]
        for i in range(0, len(nums)):
            if sum < 0:
                sum = nums[i]
            else:
                sum += nums[i]
            if sum > max_sum:
                max_sum = sum
        return max_sum

File: LeetCode-Rust/53-最大子序和.rs
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut sum = 0;
        let mut max_sum = nums[0];
        for num in nums {
            if sum < 0 {
                sum = num
            } else {
                sum += num
            }
            if sum > max_sum{
                max_sum = sum
            }
        }
        return max_sum
    }
}